### PR TITLE
updateSaveStyleActions before showing the menu

### LINKED ISSUE #46596: 'Save all Styles/Save current Style' only appears after close/reopen the layer properties window
I was writing documentation for this new feature and I noticed that the options ‚Save all Styles‘/‘Save current Style‘ only appears on my system after close/reopen the layer properties window (see pictures). I don't know if this really matters, but wanted to tell you anyway.
Greetings,
Jörn
Style_Menu_1
Style_Menu_2
Bildschirmfoto von 2021-12-16 11-46-01

_Originally posted by @Gutzeit555 in https://github.com/qgis/QGIS-Documentation/issues/4398#issuecomment-995656300_

### PR DESCRIPTION
**Pull request by SebastienPeillet**
## Description

FIX #46596

Solution :

By default, both saveAction and saveMultipleAction are created.

On each menu trigger, if there is more than one style, QGIS updates the saveAction text to `Save Current Style...` and adds the saveMultipleAction to the menu.
Otherwise the saveMultipleAction is removed from the menu and the saveAction text is updated to `Save Style...`

### CODE REVIEW

File: src/gui/vector/qgsvectorlayerproperties.cpp [hunk 1]
```diff
@@ -136,18 +136,17 @@ QgsVectorLayerProperties::QgsVectorLayerProperties(
   mActionLoadStyle = menuStyle->addAction( tr( "Load Style…" ) );
   connect( mActionLoadStyle, &QAction::triggered, this, &QgsVectorLayerProperties::loadStyle );
 
+  mActionSaveStyle = menuStyle->addAction( "Save Current Style…" );
```
Comment (DelazJ) [inline src/gui/vector/qgsvectorlayerproperties.cpp:139]:
```suggestion
  mActionSaveStyle = menuStyle->addAction( tr( "Save Current Style…" ) );
```

File: src/gui/vector/qgsvectorlayerproperties.cpp [hunk 2]
```diff
@@ -136,18 +136,17 @@ QgsVectorLayerProperties::QgsVectorLayerProperties(
...
+  connect( mActionSaveStyle, &QAction::triggered, this, &QgsVectorLayerProperties::saveStyleAs );
+  mActionSaveMultipleStyles = menuStyle->addAction( tr( "Save All Styles…" ) );
```
Comment (DelazJ) [inline src/gui/vector/qgsvectorlayerproperties.cpp:141]:
```suggestion
  mActionSaveMultipleStyles = menuStyle->addAction( tr( "Save Styles…" ) );
```
The user can choose the styles to output, so let's not use "all" here (that conveys that you might output everything with no control)

File: src/gui/vector/qgsvectorlayerproperties.cpp (hunk 1 above)
Comment (SebastienPeillet) [reply src/gui/vector/qgsvectorlayerproperties.cpp]: :+1:  thank you for the review, I missed this one !

File: src/gui/vector/qgsvectorlayerproperties.cpp (hunk 2 above)
Comment (SebastienPeillet) [reply src/gui/vector/qgsvectorlayerproperties.cpp:141]: I agree !

**Comment by nyalldawson** [comment]
@SebastienPeillet I've pushed a cleaner version of this fix at https://github.com/qgis/QGIS/pull/47080 (it's a less fragile approach). Can you take a look?

File: src/gui/vector/qgsvectorlayerproperties.cpp [hunk 3]
```diff
@@ -1373,12 +1372,43 @@ void QgsVectorLayerProperties::aboutToShowStyleMenu()
   // this should be unified with QgsRasterLayerProperties::aboutToShowStyleMenu()
   QMenu *m = qobject_cast<QMenu *>( sender() );
 
+  updateSaveStyleMenuActions( m );
+  QgsDebugMsg( "D " + m->actions().at( 1 )->text() );
```
Comment (SebastienPeillet) [inline src/gui/vector/qgsvectorlayerproperties.cpp:1376]:
```suggestion
```

**Comment by SebastienPeillet** [comment]
close in profit to #47080

**Pull request closed without merging**